File: nest-contract-test-example/src/customer/customer.dto.ts
import { IsEmail, IsNotEmpty, IsOptional } from 'class-validator';

export class CustomerRequest {
  @IsNotEmpty()
  firstName: string;

  @IsOptional()
  lastName?: string;

  @IsNotEmpty()
  @IsEmail()
  email: string;
}

export type CustomerResponse = {
  name: string;
  email: string;
};

File: kotlin-contract-test-example/src/main/kotlin/com/example/kotlincontracttestexample/service/example/CustomerDto.kt
package com.example.kotlincontracttestexample.service.example

import javax.validation.constraints.NotBlank

data class CustomerRequest(
    @get:NotBlank val firstName: String,
    @get:NotBlank val lastName: String,
    @get:NotBlank val email: String
)

data class CustomerResponse(
    val name: String,
    val email: String
)
